ext/nio4r/bytebuffer.c

```diff
@@ -305,7 +305,7 @@ static VALUE NIO_ByteBuffer_read_from(VALUE self, VALUE io)
 
     buffer->position += bytes_read;
 
-    return INT2NUM(bytes_read);
+    return INT2NUM((int)bytes_read);
```

Comment:
At this point we can assume the number is positive. So, what about using something like `SIZET2NUM`?

Reply:
Sounds good! Changed it.